src/runstats.hpp

```diff
@@ -0,0 +1,46 @@
+//
+//  runstats.hpp
+//  Str8tsSolver
+//
+
+#ifndef runstats_hpp
+#define runstats_hpp
+
+#include <string>
+
+extern bool opt_trace;
+
+struct Node {
+    friend class Runstats;
+    Node(std::string name, Node * parent) : name(name), parent(parent) { };
+    ~Node();
+private:
+    std::string name;
+    std::unique_ptr<Node> first = nullptr;
```

Comment:
Here, too, I needed "include <memory>".

Reply:
I am using llvm's clang on macOS. The include files of their version of stdlib are deeply nested. So `#include <string>` also includes `memory`. This is not the case with gnu's stdlib.

Running `g++ -H -std=c++17 src/runstats.cpp` yields header includes and nesting depth:
   
 
[option-H.txt](https://github.com/malkuh/Str8ts/files/11859982/option-H.txt)

It is hard to get independent from the platform.